docs/src/main/tut/contributing.md

```diff
@@ -0,0 +1,386 @@
+[HttpClient][httpclient-scala]. All the HTTP calls are grouped by API in the
+[github4s.free.api package][api-pkg] and, since our endpoint belongs to the repository API, we'll
+add our call to [Repos][repos-api-scala]:
+
+```scala
+def listStatuses(
+    accessToken: Option[String] = None,
+    headers: Map[String, String] = Map(),
+    owner: String,
+    repo: String,
+    ref: String): M[GHResponse[List[Status]]] =
+  httpClient.get[List[Status]](accessToken, s"repos/$owner/$repo/commits/$ref/statuses", headers)
+```
+
+## Algebra
+
+Then, we need to extend the algebra for the corresponding API so it can support our endpoint.
+Because our endpoint is part of the repository API, we need to extend the
+[RepositoryOp][repos-algebra-scala] ADT:
+
+```scala
+final case class ListStatuses(
+    owner: String,
+    repo: String,
+    ref: String,
+    accessToken: Option[String] = None
+) extends RepositoryOp[GHResponse[List[Status]]]
+```
+
+We can now expose our endpoint as a [cats.free.Free](http://typelevel.org/cats/datatypes/freemonad.html):
+
+```scala
+class RepositoryOps[F[_]](implicit I: Inject[RepositoryOp, F]) {
+  // ...
+  def listStatuses(
+      owner: String,
+      repo: String,
+      ref: String,
+      accessToken: Option[String] = None
+  ): Free[F, GHResponse[List[Status]]] =
+    Free.inject[RepositoryOp, F](ListStatuses(owner, repo, ref, accessToken))
+  // ...
+}
+```
+
+This code belongs to the [github4s.free.algebra package][algebra-pkg] in the file corresponding
+to the API, here [RepositoryOps][repos-algebra-scala].
+
+## Interpreter
+
+We're now ready to make [our repository interpreter][interpreter-scala] deal with `ListStatuses`:
+
+```scala
+def repositoryOpsInterpreter: RepositoryOp ~> K = new (RepositoryOp ~> K) {
+  val repos = new Repos()
+  def apply[A](fa: RepositoryOp[A]): K[A] = Kleisli[M, Map[String, String], A] { headers =>
+    fa match {
+      // ...
+      case ListStatuses(owner, repo, ref, accessToken) =>
+        repos.listStatus(accessToken, headers, owner, repo, ref)
+      // ...
+    }
+  }
+}
+```
+
+## Wrapper
+
+Finally, we're ready to add our endpoint to the [GHRepos][apis-scala] wrapper:
+
+```scala
+class GHRepos(accessToken: Option[String] = None)(implicit O: RepositoryOps[GitHub4s]) {
+  // ...
+  def listStatuses(
+      owner: String,
+      repo: String,
+      ref: String
+  ): GHIO[GHResponse[List[Status]]] =
+    O.listStatuses(owner, repo, ref, accessToken)
+}
+```
+
+# Test
+
+Now that we've written our source code, we're ready to write the tests.
+
+## Token
+
+The first step needed to run the tests is a valid token which we can provide through an
+environment variable:
+
+```bash
+export GITHUB4S_ACCESS_TOKEN=aaaa
+```
+
+## Integration tests
+
+The integration tests are grouped by API in [github4s.integration package][integ-pkg]. As a result,
+we'll be writing our tests in [GHReposSpec][repos-integ-spec-scala]:
```

Comment:
the funny thing is that I was planning on adding it but completely forgot about it :smile: